osu.Game.Rulesets.Catch/Objects/Drawables/CaughtObjectContainer.cs

```diff
@@ -0,0 +1,232 @@
+        }
+
+        public void DropStackedObjects(Action<DrawableCaughtObject> applyTransforms, int mirrorDirection)
+        {
+            dropStackedObjects(Clock.CurrentTime, applyTransforms, mirrorDirection);
+        }
+
+        protected override bool CheckChildrenLife()
+        {
+            bool aliveChanged = base.CheckChildrenLife();
+            aliveChanged |= lifetimeManager.Update(Time.Current);
+            return aliveChanged;
+        }
+
+        private void entryBecameAlive(LifetimeEntry lifetimeEntry)
+        {
+            var entry = (CaughtObjectEntry)lifetimeEntry;
+
+            if (entry.State == CaughtObjectState.Stacked)
+                aliveStackedObjects.Add(entry);
+
+            var drawable = getPooledDrawable(entry.HitObject);
+            drawable.Apply(entry);
+
+            addDrawable(entry, drawable);
+        }
+
+        private void entryBecameDead(LifetimeEntry lifetimeEntry)
+        {
+            var entry = (CaughtObjectEntry)lifetimeEntry;
+
+            if (entry.State == CaughtObjectState.Stacked)
+                aliveStackedObjects.Remove(entry);
+
+            removeDrawable(entry);
+        }
+
+        private void dropStackedObjects(double time, Action<DrawableCaughtObject> applyTransforms, int mirrorDirection)
+        {
+            foreach (var entry in aliveStackedObjects)
+            {
+                entry.LifetimeEnd = time;
+
+                if (!dropEntryMap.TryGetValue(entry, out var dropEntry)) continue;
+
+                dropEntry.LifetimeStart = time;
+                dropEntry.ApplyTransforms = applyTransforms;
+                dropEntry.DropPosition = GetCurrentDropPosition(entry.PositionInStack);
+                dropEntry.MirrorDirection = mirrorDirection;
+            }
+
+            aliveStackedObjects.Clear();
+        }
+
+        private void addDrawable(CaughtObjectEntry entry, DrawableCaughtObject drawable)
+        {
+            if (entry.State == CaughtObjectState.Stacked)
+            {
+                drawable.Position = entry.PositionInStack;
+                StackedObjectContainer.Add(drawable);
+            }
+            else
+            {
+                drawable.Position = entry.DropPosition;
+                drawable.Scale *= new Vector2(entry.MirrorDirection, 1);
+                droppedObjectTarget.Add(drawable);
+            }
+
+            if (entry.ApplyTransforms != null)
+            {
+                using (drawable.BeginAbsoluteSequence(entry.LifetimeStart))
+                {
+                    entry.ApplyTransforms(drawable);
+                    entry.LifetimeEnd = drawable.LatestTransformEndTime;
+                }
+            }
+
+            drawableMap[entry] = drawable;
+        }
+
+        private void removeDrawable(CaughtObjectEntry entry)
+        {
+            if (!drawableMap.TryGetValue(entry, out DrawableCaughtObject drawable))
+                return;
```

Comment:
Can use `.Remove(entry, out var drawable)`, but again, I'd really be expecting this to never fail.